document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/handler/accounts/AccountsHandlerImpl.java

```diff
@@ -1,13 +1,55 @@
 package uk.gov.companieshouse.document.generator.accounts.handler.accounts;
 
+import static uk.gov.companieshouse.document.generator.accounts.AccountsDocumentInfoServiceImpl.MODULE_NAME_SPACE;
+
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.model.accounts.Accounts;
+import uk.gov.companieshouse.document.generator.accounts.AccountType;
+import uk.gov.companieshouse.document.generator.accounts.LinkType;
+import uk.gov.companieshouse.document.generator.accounts.exception.HandlerException;
+import uk.gov.companieshouse.document.generator.accounts.exception.ServiceException;
+import uk.gov.companieshouse.document.generator.accounts.service.AccountsService;
 import uk.gov.companieshouse.document.generator.interfaces.model.DocumentInfoResponse;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
 
 @Component
 public class AccountsHandlerImpl implements AccountsHandler  {
 
+    private static final Logger LOG = LoggerFactory.getLogger(MODULE_NAME_SPACE);
+
+    @Autowired
+    AccountsService accountsService;
+
     @Override
-    public DocumentInfoResponse getAccountsData(String resourceLink) {
+    public DocumentInfoResponse getAccountsData(String resourceLink) throws HandlerException {
+        Accounts accounts;
+
+        try {
+            accounts = accountsService.getAccounts(resourceLink);
+        } catch (ServiceException e) {
+            throw new HandlerException(e.getMessage(), e.getCause());
+        }
+
+        // TODO: The accountsType variable shall be used in the implementation of abridged logic
+        AccountType accountsType = getAccountType(accounts);
+
         return new DocumentInfoResponse();
     }
+
+
+    /**
+     * Get the account type from the account link within links
+     *
+     * @return accountsData - accounts data
+     */
+    private AccountType getAccountType(Accounts accountsData) throws HandlerException {
```

Comment:
Should this method be part of the `AccountType` enum?

Reply:
There is already a method that gets the account type from the enum, this one specifically gets the account type from the accounts data set. Both do different things

Reply:
To be a bit more specific.
- The `getAccountType` in the enum simply gets the account type from the enum given the key
- The `getAccountType` in the Handler gets the account type from the accounts data set by iterating over the enum `getAccountType` for each element until it matches

Reply:
That's fine, but it feels as though it would be the concern of the enum to return an AccountType for any possible input. The 2 methods could both exist in the enum, and then be javadoc'd appropriately.

Reply:
I'm not sure, enums normally only concern themselves with the logic needed internally, they shouldn't really be connecting other working classes of the application to it. Additionally as we are currently looking through the accounts data, would we have to create a new method for a different type of accounts? This then gets larger and larger etc